File: election-backend/src/main/java/com/example/electionbackend/interfaces/IMetadataService.java
package com.example.electionbackend.interfaces;

import com.example.electionbackend.model.Metadata;
import com.example.electionbackend.type.RegionType;

import java.util.List;

/**
 * Interface for Metadata service operations.
 */
public interface IMetadataService {
    List<Metadata> getByRegionTypeAndId(RegionType regionType, String regionId);
    List<Metadata> getByRegionType(RegionType regionType);
}

File: election-frontend/src/interfaces/IMetadataService.ts
import {RegionType} from "@/enums/RegionType";
import {IMetadata} from "@/interfaces/IMetadata";

/**
 * Interface for metadata service.
 */
export interface IMetadataService {
    getByRegionTypeAndId(regionType: RegionType, regionId: string): Promise<IMetadata | null>;
}